### PR TITLE
Support Mongoid 7

### PR DESCRIPTION
**Pull request by madAle**
This would be a quick solution for supporting Mongoid 7

#207

### CODE REVIEW

**Comment by madAle** [comment]
@dblock I've made fixes to support older versions, but there's still the "danger" build that fails without much information.

I've also removed support for Mongoid < 3.1 because it wasn't run by CI anyway, so...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ### 4.0.1 (Next)
 
-* Your contribution here.
```
Comment (dblock) [inline CHANGELOG.md]: Danger is complaining about this, put it back here please for the next person.

File: spec/unit/associations_spec.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@
 
   describe Site do
     if Mongoid::Compatibility::Version.mongoid6_or_newer?
-      it { is_expected.to have_many(:users).as_inverse_of(:site).ordered_by(:email.desc).with_counter_cache }
+      it { is_expected.to have_many(:users).as_inverse_of(:site).ordered_by(:email.desc) }
```
Comment (dblock) [inline spec/unit/associations_spec.rb]: I don't think I understand this change, care to explain? Thanks.

**Comment by dblock** [comment]
Cool, almost there.

File: lib/matchers/associations.rb [hunk 3]
```diff
@@ -222,7 +240,7 @@ def matches?(actual)
           end
 
           if @association[:index]
-            if metadata.index != true
+            if !metadata.indexed?
```
Comment (dblock) [inline lib/matchers/associations.rb]: `unless metadata.indexed?`

Comment (madAle) [reply lib/matchers/associations.rb]: Yeah that would be great... unless Rubocop 😅 ->  https://travis-ci.org/mongoid/mongoid-rspec/jobs/364198957

File: spec/unit/associations_spec.rb (hunk 2 above)
Comment (madAle) [reply spec/unit/associations_spec.rb]: When running specs against Mongoid 7, it complains about the counter_cache option that's not valid. Actually counter_cache should be placed on the belongs_to side of the relationship, doesn't it?

Comment (madAle) [reply spec/unit/associations_spec.rb]: @dblock  In 3e80d95 I've fixed counter_cache testing in every mongoid version. Does it look good?

File: lib/matchers/associations.rb (hunk 3 above)
Comment (dblock) [reply lib/matchers/associations.rb]: Typical workflow for rubocop is `rubocop -a ; rubocop --auto-gen-config`.

File: spec/unit/associations_spec.rb [hunk 4]
```diff
@@ -39,8 +39,10 @@
   end
 
   describe Site do
-    if Mongoid::Compatibility::Version.mongoid6_or_newer?
+    if Mongoid::Compatibility::Version.mongoid6?
```
Comment (dblock) [inline spec/unit/associations_spec.rb]: This should be `Mongoid::Compatibility::Version.mongoid6_or_older?` to match the code in `site.rb`, right?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ### 4.0.1 (Next)
 
+* [#212](https://github.com/mongoid/mongoid-rspec/pull/212): Support mongoid 7 - [@madAle](https://github.com/madAle).
```
Comment (dblock) [inline CHANGELOG.md]: Nitpick: for 6 we said "Mongoid 6 support" below, my OCD tells me this should say "Mongoid 7 support" ;)

**Comment by dblock** [comment]
Might be a good idea to also add a compatibility section to README.

File: lib/matchers/associations.rb (hunk 3 above)
Comment (madAle) [reply lib/matchers/associations.rb]: 👍🏻

File: spec/unit/associations_spec.rb (hunk 4 above)
Comment (madAle) [reply spec/unit/associations_spec.rb]: Yeah, there's a mismatch, but it should be  ```Mongoid::Compatibility::Version.mongoid6``` on both files

File: CHANGELOG.md (hunk 5 above)
Comment (madAle) [reply CHANGELOG.md]: 👍

**Comment by madAle** [comment]
@dblock Done

**Comment by dblock** [comment]
Looks great, I merged this. Maybe @rodrigopinto or other maintainers want to cut a release?

**Comment by madAle** [comment]
Thanks.

**Comment by gurix** [comment]
A release for 4.0.1 or so would be nice.

**Pull request merged**